travis_configure.sh

```diff
@@ -15,27 +15,30 @@ print_error_msg ()
 {
     test ! -z "$1" || exit 1
     echo "$1"
-    echo "  os:      $TRAVIS_OS_NAME"
-    echo "  package: $DIST_PACKAGE_TARGET"
+    echo "  os:         $TRAVIS_OS_NAME"
+    echo "  lsb:        $DIST_NAME $DIST_VERSION"
+    echo "  targets:    $MAKE_TARGET"
 
     return 0
 }
 
 # Script which is run before the installation script is called
 #
-# Installs dependencies
-# linux:
-#   deb:
-#     - fpm
-# osx:
+# For linux based builds docker-compose is used to set up the build environment
+#
+# For osx based builds homebrew is used to install required packages:
 #   - sed, (libtool), openssl, mysql
 # Returns 0
 before_install()
 {
     if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
-        if [[ "$DIST_PACKAGE_TARGET" == "DEB" ]]; then
-            gem install fpm -v 1.8.1
+        # build images
+        docker-compose -f ./docker-compose.yml \
+            -f ./docker/docker-compose.$DIST_NAME.$DIST_VERSION.yml \
```

Comment:
OK, I see now, nice trick. I think the 3 last commits (including this one) should be squashed into one.

Reply:
Sure, guess I was more focused on the extent than the logics of the changes.